diff -Nur CheckM-1.0.13.orig/bin/checkm CheckM-1.0.13/bin/checkm
--- CheckM-1.0.13.orig/bin/checkm	2019-01-07 21:04:48.000000000 +0100
+++ CheckM-1.0.13/bin/checkm	2019-06-14 15:32:39.124664899 +0200
@@ -33,6 +33,8 @@
 import sys
 import os
 
+sys.path.append('PREFIX/lib/python2.7/site-packages')
+
 from checkm import main
 from checkm.defaultValues import DefaultValues
 from checkm.customHelpFormatter import CustomHelpFormatter
