diff -Nur CheckM-1.2.2.orig/checkm/checkmData.py CheckM-1.2.2/checkm/checkmData.py
--- CheckM-1.2.2.orig/checkm/checkmData.py	2022-10-31 14:52:47.000000000 +0100
+++ CheckM-1.2.2/checkm/checkmData.py	2023-01-02 15:32:10.946857381 +0100
@@ -50,7 +50,10 @@
             with open(self.configFile, 'r') as local_config:
                 # config is a one line file
                 for line in local_config:
-                    return json.loads(line)
+                    ret = json.loads(line)
+                    if not os.path.exists(ret["dataRoot"]) and os.path.exists(os.environ['HOME']+ "/checkm"):
+                        ret["dataRoot"] = os.environ['HOME']+ "/checkm"
+                    return ret
 
         except Exception:
             print("There seems to be a problem with loading the CheckM config file")
