--- tigrfam_search.py.orig	2022-11-03 11:01:57.317558429 +0100
+++ tigrfam_search.py	2022-11-03 11:02:52.409232808 +0100
@@ -138,7 +138,7 @@
                 args = ['hmmsearch', '-o', hmmsearch_out, '--tblout', output_hit_file,
                         '--noali', '--notextw', '--cut_nc', '--cpu',
                         str(self.cpus_per_genome), self.tigrfam_hmms, gene_file]
-                p = subprocess.Popen(args, stdout=subprocess.PIPE, encoding='utf-8')
+                p = subprocess.Popen(args, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
                 stdout, stderr = p.communicate()
 
                 if p.returncode != 0:
