--- hmmer-3.1b2/src/hmmbuild.c	2015-02-26 13:46:03.000000000 +0100
+++ hmmer-3.1b2.p/src/hmmbuild.c	2017-07-27 15:31:47.151256000 +0200
@@ -997,7 +997,6 @@
 static void
 serial_loop(WORKER_INFO *info, struct cfg_s *cfg, const ESL_GETOPTS *go)
 {
-  P7_BUILDER *bld         = NULL;
   ESL_MSA    *msa         = NULL;
   ESL_SQ     *sq          = NULL;
   ESL_MSA    *postmsa     = NULL;
@@ -1019,13 +1018,13 @@
 
       /*         bg   new-HMM trarr gm   om  */
       if ( msa->nseq == 1 && esl_opt_IsUsed(go, "--singlemx")) {
-        if ((status = esl_sq_FetchFromMSA(msa, 0, &sq)) != eslOK) p7_Fail("build failed: %s", bld->errbuf);
-        if ((status = p7_SingleBuilder(info->bld, sq, info->bg, &hmm, NULL, NULL, NULL)) != eslOK) p7_Fail("build failed: %s", bld->errbuf);
+        if ((status = esl_sq_FetchFromMSA(msa, 0, &sq)) != eslOK) p7_Fail("build failed: %s", info->bld->errbuf);
+        if ((status = p7_SingleBuilder(info->bld, sq, info->bg, &hmm, NULL, NULL, NULL)) != eslOK) p7_Fail("build failed: %s", info->bld->errbuf);
         esl_sq_Destroy(sq);
         sq = NULL;
         hmm->eff_nseq = 1;
       } else {
-        if ((status = p7_Builder(info->bld, msa, info->bg, &hmm, NULL, NULL, NULL, postmsa_ptr)) != eslOK) p7_Fail("build failed: %s", bld->errbuf);
+        if ((status = p7_Builder(info->bld, msa, info->bg, &hmm, NULL, NULL, NULL, postmsa_ptr)) != eslOK) p7_Fail("build failed: %s", info->bld->errbuf);
 
         //if not --singlemx, but the user set the popen/pextend flags, override the computed gap params now:
         if (info->bld->popen != -1 || info->bld->pextend != -1) {
