diff -Nur hmmer-3.1b2/Makefile.in hmmer-3.1b2.strip/Makefile.in
--- hmmer-3.1b2/Makefile.in	2016-08-05 17:52:34.400826120 +0200
+++ hmmer-3.1b2.strip/Makefile.in	2016-08-05 17:48:37.013164056 +0200
@@ -95,7 +95,7 @@
 	nhmmscan\
 	hmmpgmd
 
-.PHONY: all dev check pdf install uninstall clean distclean TAGS
+.PHONY: all dev check pdf install install-strip uninstall clean distclean TAGS
 
 # all: Compile all documented executables.
 #      (Excludes test programs.)
@@ -148,6 +148,15 @@
 	${QUIET_SUBDIR0}src           ${QUIET_SUBDIR1} install
 	${QUIET_SUBDIR0}documentation ${QUIET_SUBDIR1} install
 
+install-strip: 
+	if [ ! -d ${DESTDIR}${bindir} ]; then ${INSTALL} -d ${DESTDIR}${bindir} ; fi
+	if [ ! -d ${DESTDIR}${libdir} ]; then ${INSTALL} -d ${DESTDIR}${libdir} ; fi
+	if [ ! -d ${DESTDIR}${includedir} ]; then  ${INSTALL} -d ${DESTDIR}${includedir} ; fi
+	if [ ! -d ${DESTDIR}${man1dir} ]; then  ${INSTALL} -d ${DESTDIR}${man1dir} ; fi
+	if [ ! -d ${DESTDIR}${pdfdir} ]; then  ${INSTALL} -d ${DESTDIR}${pdfdir} ; fi
+	${QUIET_SUBDIR0}src           ${QUIET_SUBDIR1} install-strip
+	${QUIET_SUBDIR0}documentation ${QUIET_SUBDIR1} install
+
 # uninstall: Reverses the steps of "make install".
 #
 uninstall: 
diff -Nur hmmer-3.1b2/src/Makefile.in hmmer-3.1b2.strip/src/Makefile.in
--- hmmer-3.1b2/src/Makefile.in	2015-02-26 13:46:04.000000000 +0100
+++ hmmer-3.1b2.strip/src/Makefile.in	2016-08-05 17:48:50.920983353 +0200
@@ -226,7 +226,7 @@
 		        ${MAKE} -C $$subdir
 endif
 
-.PHONY: all dev tests check install uninstall distclean clean TAGS
+.PHONY: all dev tests check install install-strip uninstall distclean clean TAGS
 
 all:   ${PROGS} .FORCE
 
@@ -332,6 +332,18 @@
 	done
 	${INSTALL} -m 0644 p7_config.h ${DESTDIR}${includedir}/ ;\
 
+install-strip:
+	${QUIET_SUBDIR0}${IMPLDIR} ${QUIET_SUBDIR1} install
+	for file in ${PROGS}; do \
+	   ${INSTALL} -s -m 0755 $$file ${DESTDIR}${bindir}/ ;\
+	done
+	${INSTALL} -m 0755 libhmmer.a ${DESTDIR}${libdir}/
+	for file in ${HDRS}; do \
+	   ${INSTALL} -m 0644 ${srcdir}/$$file ${DESTDIR}${includedir}/ ;\
+	done
+	${INSTALL} -m 0644 p7_config.h ${DESTDIR}${includedir}/ ;\
+
+
 uninstall:
 	${QUIET_SUBDIR0}${IMPLDIR} ${QUIET_SUBDIR1} uninstall
 	for file in ${PROGS}; do \
