diff -Nur metacv_2_3_0.orig/Makefile metacv_2_3_0/Makefile
--- metacv_2_3_0.orig/Makefile	2013-01-22 08:18:23.000000000 +0100
+++ metacv_2_3_0/Makefile	2017-03-05 19:29:00.911169000 +0100
@@ -1,15 +1,15 @@
-CC        = g++
-COMPILE.C = $(CC) -c
+CXX        = g++
+COMPILE.C = $(CXX) -c
 VPATH     = ./source_code/
-MAKEEXE   = $(CC) 
+MAKEEXE   = $(CXX) 
 
 OBJ       = metacv.o
 EXE       = metacv
 
 $(EXE):   $(OBJ)
-	$(MAKEEXE) $(OBJ) -o $(EXE) -lpthread
+	$(MAKEEXE) $(CXXFLAGS) $(OBJ) -o $(EXE) -lpthread
 $(OBJ):   $(VPATH)MetaCV.cpp
-	$(COMPILE.C) $(VPATH)MetaCV.cpp -o $(OBJ)
+	$(COMPILE.C) $(CXXFLAGS) $(VPATH)MetaCV.cpp -o $(OBJ)
 
 clean:
 	rm -f $(OBJ) $(EXE)
