diff -Nur metacv_2_3_0.orig/source_code/MetaCV.cpp metacv_2_3_0/source_code/MetaCV.cpp
--- metacv_2_3_0.orig/source_code/MetaCV.cpp	2013-01-22 09:00:40.000000000 +0100
+++ metacv_2_3_0/source_code/MetaCV.cpp	2017-03-05 19:24:05.568712000 +0100
@@ -3550,7 +3550,7 @@
     is.open(filename.c_str(), ios::in);
     if (is.fail()) {
         fprintf(stderr, "warning[24] (not exist): %s\n", filename.c_str());
-        fprintf(stderr, "checking for taxa id\n", filename.c_str());
+        fprintf(stderr, "checking for taxa id\n");
         is.close();
 
         StringTokenizer nizer(filename, ",");
@@ -3628,7 +3628,7 @@
     is.open(filename.c_str(), ios::in);
     if (is.fail()) {
         fprintf(stderr, "warning[24] (not exist): %s\n", filename.c_str());
-        fprintf(stderr, "checking for taxa id\n", filename.c_str());
+        fprintf(stderr, "checking for taxa id\n");
         is.close();
         return;
     }
