diff -Nur mothur-1.37.4.orig/Makefile mothur-1.37.4/Makefile
--- mothur-1.37.4.orig/Makefile	2016-05-11 17:40:31.000000000 +0200
+++ mothur-1.37.4/Makefile	2016-07-08 17:20:04.861992040 +0200
@@ -18,7 +18,8 @@
 VERSION = "\"1.37.1\""
 
 ifeq  ($(strip $(64BIT_VERSION)),yes)
-    CXXFLAGS += -DBIT_VERSION
+    CXXFLAGS += -DBIT_VERSION -m64
+    LDFLAGS += -m64
 endif
 
 # Fastest
@@ -79,7 +80,7 @@
     OBJECTS+=$(patsubst %.c,%.o,$(wildcard *.c))
 
 mothur : $(OBJECTS) uchime
-	$(CXX) $(LDFLAGS) $(TARGET_ARCH) -o $@ $(OBJECTS) $(LIBS)
+	$(CXX) -static-libstdc++ -static-libgcc $(LDFLAGS) $(TARGET_ARCH) -o $@ $(OBJECTS) $(LIBS)
 
 uchime:
 	cd source/uchime_src && ./mk && mv uchime ../../ && cd ..
diff -Nur mothur-1.37.4.orig/source/mothur.h mothur-1.37.4/source/mothur.h
--- mothur-1.37.4.orig/source/mothur.h	2016-05-11 17:40:31.000000000 +0200
+++ mothur-1.37.4/source/mothur.h	2016-07-08 17:20:04.868364018 +0200
@@ -65,7 +65,9 @@
 	#include <sys/resource.h>
 	#include <sys/types.h>
 	#include <sys/stat.h>
-    #include <sys/sysctl.h>
+#if defined (linux) || (__linux) || (__linux__)
+        #include <sys/sysctl.h>
+#endif
 	#include <unistd.h>
 
 	
diff -Nur mothur-1.37.4.orig/source/mothurout.cpp mothur-1.37.4/source/mothurout.cpp
--- mothur-1.37.4.orig/source/mothurout.cpp	2016-05-11 17:40:31.000000000 +0200
+++ mothur-1.37.4/source/mothurout.cpp	2016-07-08 17:20:04.877661119 +0200
@@ -596,7 +596,7 @@
 /***********************************************************************/
 unsigned long MothurOut::get_phys_pages () {
     unsigned long phys_pages = 0;
-#if (_SC_PAGE_SIZE)
+#if (_SC_PAGE_SIZE) && ((linux) || (__linux) || (__linux__))
     uint64_t mem;
     size_t len = sizeof(mem);
     sysctlbyname("hw.memsize", &mem, &len, NULL, 0);
diff -Nur mothur-1.37.4.orig/source/uchime_src/mk mothur-1.37.4/source/uchime_src/mk
--- mothur-1.37.4.orig/source/uchime_src/mk	2016-05-11 17:40:31.000000000 +0200
+++ mothur-1.37.4/source/uchime_src/mk	2016-07-08 17:21:39.548715389 +0200
@@ -7,7 +7,7 @@
 for CPPName in $CPPNames
 do
   echo $CPPName >&2
-  g++ $ENV_GCC_OPTS -c -D_FILE_OFFSET_BITS=64 -DNDEBUG=1 -DUCHIMES=1 $CPPName.cpp -o $CPPName.o  >> mk.stdout 2>> tmp.stderr
+  g++ $ENV_GCC_OPTS -m64 -c -D_FILE_OFFSET_BITS=64 -DNDEBUG=1 -DUCHIMES=1 $CPPName.cpp -o $CPPName.o  >> mk.stdout 2>> tmp.stderr
   cat tmp.stderr
 	cat tmp.stderr >> mk.stderr
 	rm -f tmp.stderr
@@ -17,7 +17,7 @@
 #if [ `uname -s` == Linux ] ; then
 #    LINK_OPTS=-static
 #fi
-g++ $LINK_OPTS $ENV_LINK_OPTS -g -o uchime $ObjNames >> mk.stdout 2>> tmp.stderr
+g++ -m64 -static-libstdc++ -static-libgcc $LINK_OPTS $ENV_LINK_OPTS -g -o uchime $ObjNames >> mk.stdout 2>> tmp.stderr
 cat tmp.stderr
 cat tmp.stderr >> mk.stderr
 rm -f tmp.stderr
diff -Nur mothur-1.37.4.orig/source/uchime_src/myutils.cpp mothur-1.37.4/source/uchime_src/myutils.cpp
--- mothur-1.37.4.orig/source/uchime_src/myutils.cpp	2016-05-11 17:40:31.000000000 +0200
+++ mothur-1.37.4/source/uchime_src/myutils.cpp	2016-07-08 17:20:04.890764820 +0200
@@ -721,12 +721,12 @@
 	{
 	int HH = Secs/3600;
 	int MM = (Secs - HH*3600)/60;
-	int SS = Secs%60;
+	int SS6 = Secs%60;
 	static char Str[16];
 	if (HH == 0)
-		sprintf(Str, "%02d:%02d", MM, SS);
+		sprintf(Str, "%02d:%02d", MM, SS6);
 	else
-		sprintf(Str, "%02d:%02d:%02d", HH, MM, SS);
+		sprintf(Str, "%02d:%02d:%02d", HH, MM, SS6);
 	return Str;
 	}
 
