diff -Nur Rserve.orig/MD5 Rserve/MD5
--- Rserve.orig/MD5	2013-08-21 23:35:21.000000000 +0200
+++ Rserve/MD5	2018-05-04 09:48:42.762187677 +0200
@@ -89,7 +89,7 @@
 b2c9af5ad6e9685f0e2bf4a495e18e12 *man/Rserv.Rd
 ffcadeb5fc4e139fa5da05eac3c3d905 *man/run.Rserve.Rd
 e76ccbbe06bdf6737fe05d45c953dc53 *man/self.Rd
-b52eaae46f8d1b4bc77b62a9a9368c90 *src/Makevars.in
+3d71db0686624f4c54711319b1bbbd3f *src/Makevars.in
 090d4ac765eb970eadfc5787be2f39ab *src/Makevars.win
 fa8a1c083d32e1771ade42528af860bf *src/RSserver.c
 d4da9b05f710d31f0e22fc07de913827 *src/RSserver.h
diff -Nur Rserve.orig/src/Makevars.in Rserve/src/Makevars.in
--- Rserve.orig/src/Makevars.in	2013-08-21 23:09:33.000000000 +0200
+++ Rserve/src/Makevars.in	2018-05-04 09:48:13.596961606 +0200
@@ -1,4 +1,4 @@
-PKG_CPPFLAGS=-DRSERVE_PKG -I. -Iinclude @CPPFLAGS@
+PKG_CPPFLAGS=-DRSERVE_PKG -I. -Iinclude @CPPFLAGS@ -D_XPG4_2 -D__EXTENSIONS__
 PKG_LIBS=@LIBS@
 
 all: $(SHLIB) @WITH_SERVER_TRUE@ server
