diff -Nur trinityrnaseq-v2.12.0.orig/Chrysalis/analysis/Chrysalis.cc trinityrnaseq-v2.12.0/Chrysalis/analysis/Chrysalis.cc
--- trinityrnaseq-v2.12.0.orig/Chrysalis/analysis/Chrysalis.cc	2021-03-05 02:22:54.000000000 +0100
+++ trinityrnaseq-v2.12.0/Chrysalis/analysis/Chrysalis.cc	2021-05-26 11:38:23.790439642 +0200
@@ -258,7 +258,7 @@
     commandArg<string> iwormStringCmmd("-iworm","inchworm file", "");
     commandArg<string> oStringCmmd("-o","output directory");
     commandArg<bool> pairsStringCmmd("-paired", "paired-end reads are used.", false);
-    commandArg<string> butterflyCmmd("-butterfly","butterfly executable", "../Butterfly/Butterfly.jar");
+    commandArg<string> butterflyCmmd("-butterfly","butterfly executable", "../share/Butterfly.jar");
     commandArg<bool> skipCmmd("-skip","skip initial 2 steps", false);
     commandArg<bool> strandCmmd("-strand","strand-specific data", false);
     commandArg<bool> nobreakCmmd("-nobreak","skip breaking", false);
diff -Nur trinityrnaseq-v2.12.0.orig/Trinity trinityrnaseq-v2.12.0/Trinity
--- trinityrnaseq-v2.12.0.orig/Trinity	2021-03-05 02:21:01.000000000 +0100
+++ trinityrnaseq-v2.12.0/Trinity	2021-05-26 11:38:23.794439618 +0200
@@ -6,7 +6,7 @@
 no strict qw(subs refs);
 
 use FindBin;
-use lib ("$FindBin::RealBin/PerlLib");
+use lib ("$FindBin::RealBin/../share/perl/");
 use File::Basename;
 use Time::localtime;
 use Cwd;
@@ -36,15 +36,15 @@
 
 #directory defnintions
 my $ROOTDIR = "$FindBin::RealBin";
-my $UTILDIR = "$ROOTDIR/util";
+my $UTILDIR = "$ROOTDIR/../share/util";
 my $MISCDIR = "$UTILDIR/misc";
-my $INCHWORM_DIR = "$ROOTDIR/Inchworm/bin/";
-my $CHRYSALIS_DIR = "$ROOTDIR/Chrysalis";
-my $BUTTERFLY_DIR = "$ROOTDIR/Butterfly";
+my $INCHWORM_DIR = "$ROOTDIR";
+my $CHRYSALIS_DIR = "$ROOTDIR";
+my $BUTTERFLY_DIR = "$ROOTDIR";
 my $COLLECTL_DIR = "$ROOTDIR/trinity-plugins/COLLECTL/collectl";
 my $TRINITY_PLUGINS_DIR = "$ROOTDIR/trinity-plugins";
 my $TRINITY_BIN_DIR = "$TRINITY_PLUGINS_DIR/BIN";
-my $PARAFLY = "$TRINITY_BIN_DIR/ParaFly";
+my $PARAFLY = "$ROOTDIR/ParaFly";
 my $HTSLIB_DIR = "$ROOTDIR/trinity-plugins/htslib";
 
 $ENV{LD_LIBRARY_PATH} .= ":${HTSLIB_DIR}";
@@ -56,7 +56,7 @@
 my $TRIMMOMATIC = "$ROOTDIR/trinity-plugins/Trimmomatic/trimmomatic.jar";
 my $TRIMMOMATIC_DIR = "$ROOTDIR/trinity-plugins/Trimmomatic";
 
-$ENV{PATH} = "$ROOTDIR/trinity-plugins/BIN:$ENV{PATH}";
+$ENV{PATH} = "$ROOTDIR:PREFIX/bin:$ENV{PATH}";
 
 my $JAVA_VERSION_REQUIRED = 8;
 
@@ -1022,7 +1022,7 @@
 }
 
 unless ($BFLY_JAR) {
-    $BFLY_JAR = "$BUTTERFLY_DIR/Butterfly.jar";
+    $BFLY_JAR = "$BUTTERFLY_DIR/../share/Butterfly.jar";
 }
 
 
@@ -1933,7 +1933,7 @@
         
         my $pipeliner = new Pipeliner(-verbose => ($TRINITY_COMPLETE_FLAG) ? $VERBOSE : max(1, $VERBOSE));
                 
-        my $cmd = "$ROOTDIR/Analysis/SuperTranscripts/Trinity_gene_splice_modeler.py --trinity_fasta $Trinity_fasta_file --out_prefix $Trinity_fasta_file.ST --incl_cdna";
+        my $cmd = "$ROOTDIR/../share/Analysis/SuperTranscripts/Trinity_gene_splice_modeler.py --trinity_fasta $Trinity_fasta_file --out_prefix $Trinity_fasta_file.ST --incl_cdna";
         $pipeliner->add_commands( new Command($cmd, "$polish_checkpoints_dir/trinity_st.ok"));
         
         $pipeliner->run();
@@ -2066,7 +2066,7 @@
             ## generate the scaffold info
                         
             if ($USE_PERL_SCAFFOLDER) {
-                $cmd = "$ROOTDIR/util/support_scripts/scaffold_iworm_contigs.pl $bowtie_sam_file $inchworm_file > $iworm_scaffolds_file";
+                $cmd = "$ROOTDIR/../share/util/support_scripts/scaffold_iworm_contigs.pl $bowtie_sam_file $inchworm_file > $iworm_scaffolds_file";
             }
             else {
                 # this c++ code needs to be updated to reflect the new clustering criteria
@@ -2131,7 +2131,7 @@
         
         my $welds_to_cluster_file = "$chrysalis_output_dir/iworm_cluster_welds_graph.txt";
         
-        my $graphFromFasta_cmd = "$CHRYSALIS_DIR/bin/GraphFromFasta -i $inchworm_file -r $inchworm_target_fa -min_contig_length $min_contig_length -min_glue $min_glue -glue_factor $glue_factor -min_iso_ratio $min_iso_ratio -t $CPU -k " . ($KMER_SIZE-1) . " -kk $GRAPH_FROM_FASTA_KK $GRAPH_FROM_FASTA_CUSTOM_PARAMS";
+        my $graphFromFasta_cmd = "$CHRYSALIS_DIR/GraphFromFasta -i $inchworm_file -r $inchworm_target_fa -min_contig_length $min_contig_length -min_glue $min_glue -glue_factor $glue_factor -min_iso_ratio $min_iso_ratio -t $CPU -k " . ($KMER_SIZE-1) . " -kk $GRAPH_FROM_FASTA_KK $GRAPH_FROM_FASTA_CUSTOM_PARAMS";
        
         if ($SS_lib_type) {
             $graphFromFasta_cmd .= " -strand ";
@@ -2160,7 +2160,7 @@
         
         ## now cluster into chrysalis components:
         
-        my $bubble_cluster_cmd = "$CHRYSALIS_DIR/bin/BubbleUpClustering -i $inchworm_file  -weld_graph $welds_to_cluster_file_sorted -min_contig_length $min_contig_length -max_cluster_size $MAX_CHRYSALIS_CLUSTER_SIZE ";
+        my $bubble_cluster_cmd = "$CHRYSALIS_DIR/BubbleUpClustering -i $inchworm_file  -weld_graph $welds_to_cluster_file_sorted -min_contig_length $min_contig_length -max_cluster_size $MAX_CHRYSALIS_CLUSTER_SIZE ";
 
         if ($CHRYSALIS_DEBUG_WELD_ALL) {
             $bubble_cluster_cmd .= " -debug_weld_all ";
@@ -2180,7 +2180,7 @@
         ## create iworm bundles
         
         
-        my $cmd = "$CHRYSALIS_DIR/bin/CreateIwormFastaBundle -i $graphFromFasta_outfile -o $iworm_bundles_fasta_file -min $min_contig_length";
+        my $cmd = "$CHRYSALIS_DIR/CreateIwormFastaBundle -i $graphFromFasta_outfile -o $iworm_bundles_fasta_file -min $min_contig_length";
         
         
         $chrysalis_pipeliner->add_commands( new Command($cmd, "$iworm_bundles_fasta_file.ok"));
@@ -2192,7 +2192,7 @@
     { 
         ## map reads to chrysalis components:
                         
-        my $cmd .= "$CHRYSALIS_DIR/bin/ReadsToTranscripts -i $chrysalis_reads_fa -f $iworm_bundles_fasta_file -o $reads_to_components_output_file -t $CPU -max_mem_reads $max_reads_per_loop ";
+        my $cmd .= "$CHRYSALIS_DIR/ReadsToTranscripts -i $chrysalis_reads_fa -f $iworm_bundles_fasta_file -o $reads_to_components_output_file -t $CPU -max_mem_reads $max_reads_per_loop ";
         if ($SS_lib_type) {
             $cmd .= " -strand ";
         }
@@ -2303,7 +2303,7 @@
 
             { # quantify graph command
                 
-                my $quantify_graph_cmd = "$CHRYSALIS_DIR/bin/QuantifyGraph -g $base_filename.graph.tmp "
+                my $quantify_graph_cmd = "$CHRYSALIS_DIR/QuantifyGraph -g $base_filename.graph.tmp "
                     . " -i $base_filename.reads.tmp "
                     . " -o $base_filename.graph.out "
                     . " -max_reads $max_reads_per_graph "
@@ -2559,7 +2559,7 @@
 
         my $jelly_db = "mer_counts.$kmer_size.$SR_flag.jf";        
         
-        my $cmd = "jellyfish count -t $CPU -m $kmer_size -s $jelly_hash_size -o $jelly_db";
+        my $cmd = "jellyfish-2 count -t $CPU -m $kmer_size -s $jelly_hash_size -o $jelly_db";
         
         unless ($SS_lib_type) {
             ## count both strands
@@ -2572,12 +2572,12 @@
                 
 
         
-        $cmd = "jellyfish dump -L $min_kmer_cov $jelly_db > $jelly_kmer_fa_file";
+        $cmd = "jellyfish-2 dump -L $min_kmer_cov $jelly_db > $jelly_kmer_fa_file";
         
         $pipeliner->add_commands(new Command($cmd, ".jellyfish_dump.$kmer_size.mincov$min_kmer_cov.$SR_flag.ok") );        
         
         ## write a histogram of the kmer counts.
-        $cmd = "jellyfish histo -t $CPU -o $jelly_kmer_fa_file.histo $jelly_db";
+        $cmd = "jellyfish-2 histo -t $CPU -o $jelly_kmer_fa_file.histo $jelly_db";
         
         $pipeliner->add_commands( new Command($cmd, ".jellyfish_histo.$kmer_size.mincov$min_kmer_cov.$SR_flag.ok") );
         
@@ -3178,7 +3178,7 @@
         #my $cmd = "cd $collectl_output_directory && exec ${COLLECTL_DIR}/collectl $collectl_param --procfilt u$collectl_userid -f $collectl_output_directory/y";
         
         
-        my $cmd = "bash -c \"set -eou pipefail && cd $collectl_output_directory && exec nohup ${COLLECTL_DIR}/collectl --procopts w  --procfilt u$collectl_userid --interval :$COLLECTL_INTERVAL_SECONDS -sZ -oD | egrep 'Trinity|trinityrnaseq|Inchworm|Chrysalis|Butterfly|jellyfish|samtools|sort|bowtie' | egrep -v 'egrep' > $collectl_output_directory/collectl.dat \" ";
+        my $cmd = "bash -c \"set -eou pipefail && cd $collectl_output_directory && exec nohup ${COLLECTL_DIR}/collectl --procopts w  --procfilt u$collectl_userid --interval :$COLLECTL_INTERVAL_SECONDS -sZ -oD | egrep 'Trinity|trinityrnaseq|Inchworm|Chrysalis|Butterfly|jellyfish-2|samtools|sort|bowtie' | egrep -v 'egrep' > $collectl_output_directory/collectl.dat \" ";
                 
         
         ## fork a child to run collectl
@@ -4003,11 +4003,11 @@
     }
 
     # jellyfish
-    my $jellyfish_path = `which jellyfish`;
+    my $jellyfish_path = `which jellyfish-2`;
     if ($jellyfish_path =~ /\w/) {
         print "Found jellyfish at: $jellyfish_path\n" if $VERBOSE;
 
-        my $version_info = `jellyfish --version`;
+        my $version_info = `jellyfish-2 --version`;
         my ($name, $version) = split(/\s+/, $version_info);
         unless ($version =~ /^2\./) {
             die "Error, need jellyfish v2, instead found $version_info ; Get it here: http://www.genome.umd.edu/jellyfish.html";
diff -Nur trinityrnaseq-v2.12.0.orig/util/insilico_read_normalization.pl trinityrnaseq-v2.12.0/util/insilico_read_normalization.pl
--- trinityrnaseq-v2.12.0.orig/util/insilico_read_normalization.pl	2021-03-05 02:21:02.000000000 +0100
+++ trinityrnaseq-v2.12.0/util/insilico_read_normalization.pl	2021-05-26 11:38:23.802439570 +0200
@@ -18,7 +18,7 @@
 use COMMON;
 
 
-$ENV{PATH} = "$FindBin::Bin/../trinity-plugins/BIN:$ENV{PATH}";
+$ENV{PATH} = "$FindBin::Bin:$ENV{PATH}";
 
 
 open (STDERR, ">&STDOUT");  ## capturing stderr and stdout in a single stdout stream
@@ -117,8 +117,8 @@
     ;
 
 my $ROOTDIR = "$FindBin::RealBin/../";
-my $UTILDIR = "$ROOTDIR/util/support_scripts/";
-my $INCHWORM_DIR = "$ROOTDIR/Inchworm";
+my $UTILDIR = "$ROOTDIR/../share/util/support_scripts/";
+my $INCHWORM_DIR = "$ROOTDIR/../";
 
 unless (@ARGV) {
     die "$usage\n";
diff -Nur trinityrnaseq-v2.12.0.orig/util/misc/compare_bflies.pl trinityrnaseq-v2.12.0/util/misc/compare_bflies.pl
--- trinityrnaseq-v2.12.0.orig/util/misc/compare_bflies.pl	2021-03-05 02:21:02.000000000 +0100
+++ trinityrnaseq-v2.12.0/util/misc/compare_bflies.pl	2021-05-26 11:38:23.806439546 +0200
@@ -37,7 +37,7 @@
         my $cmd = "ln -s $comp.reads .;  ln -s $comp.out .";
         &process_cmd($cmd);
 
-        $cmd = "java -Xmx4G -jar $TRINITY_HOME/Butterfly/prev_vers/Butterfly_r2013_08_14.jar -N 100000 -L 200 -F 500 -C " . basename($comp) . " --path_reinforcement_distance=75 --max_number_of_paths_per_node=10 -V 15 --stderr 2>&1 | tee log.txt";
+        $cmd = "java -Xmx4G -jar $TRINITY_HOME/../share/Butterfly_r2013_08_14.jar -N 100000 -L 200 -F 500 -C " . basename($comp) . " --path_reinforcement_distance=75 --max_number_of_paths_per_node=10 -V 15 --stderr 2>&1 | tee log.txt";
 
         open (my $ofh, ">bfly.cmd") or die $!;
         print $ofh $cmd;
@@ -61,7 +61,7 @@
         &process_cmd($cmd);
         
 
-        $cmd = "java -Xmx4G -jar $TRINITY_HOME/Butterfly/Butterfly.jar -N 100000 -L 200 -F 500 -C " . basename($comp) . " --path_reinforcement_distance=75 -V 15 --stderr 2>&1 | tee log.txt";
+        $cmd = "java -Xmx4G -jar $TRINITY_HOME/../share/Butterfly.jar -N 100000 -L 200 -F 500 -C " . basename($comp) . " --path_reinforcement_distance=75 -V 15 --stderr 2>&1 | tee log.txt";
 
         open (my $ofh, ">bfly.cmd") or die $!;
         print $ofh $cmd;
diff -Nur trinityrnaseq-v2.12.0.orig/util/misc/sim_test_framework/run_Trinity_eval.pl trinityrnaseq-v2.12.0/util/misc/sim_test_framework/run_Trinity_eval.pl
--- trinityrnaseq-v2.12.0.orig/util/misc/sim_test_framework/run_Trinity_eval.pl	2021-03-05 02:21:05.000000000 +0100
+++ trinityrnaseq-v2.12.0/util/misc/sim_test_framework/run_Trinity_eval.pl	2021-05-26 11:38:23.810439522 +0200
@@ -17,7 +17,7 @@
 
 my $help_flag;
 my $ref_trans_fa;
-my $BFLY_JAR = "$ENV{TRINITY_HOME}/Butterfly/Butterfly.jar";
+my $BFLY_JAR = "$ENV{TRINITY_HOME}/../share/Butterfly.jar";
 my $INCLUDE_REF_TRANS = 0;
 my $OUT_DIR = "testing_dir";
 my $MIN_CONTIG_LENGTH = 200;
diff -Nur trinityrnaseq-v2.12.0.orig/util/support_scripts/write_partitioned_trinity_cmds.pl trinityrnaseq-v2.12.0/util/support_scripts/write_partitioned_trinity_cmds.pl
--- trinityrnaseq-v2.12.0.orig/util/support_scripts/write_partitioned_trinity_cmds.pl	2021-03-05 02:21:06.000000000 +0100
+++ trinityrnaseq-v2.12.0/util/support_scripts/write_partitioned_trinity_cmds.pl	2021-05-26 11:38:23.810439522 +0200
@@ -75,7 +75,7 @@
     
     my $file = pop @x;
     
-    my $cmd = "$FindBin::RealBin/../../Trinity --single \"$file\" --output \"$file.out\" $trin_args ";
+    my $cmd = "$FindBin::RealBin/../../../bin/Trinity --single \"$file\" --output \"$file.out\" $trin_args ";
     
     print "$cmd\n";
 }
