diff -Nur trinityrnaseq-v2.12.0.orig/trinity-plugins/bamsifter/Makefile trinityrnaseq-v2.12.0/trinity-plugins/bamsifter/Makefile
--- trinityrnaseq-v2.12.0.orig/trinity-plugins/bamsifter/Makefile	2021-03-05 02:22:55.000000000 +0100
+++ trinityrnaseq-v2.12.0/trinity-plugins/bamsifter/Makefile	2021-05-26 11:35:02.099651698 +0200
@@ -4,6 +4,7 @@
 
 sift_bam_max_cov: sift_bam_max_cov.cpp htslib/version.h
 	g++ -std=c++11 -o _sift_bam_max_cov sift_bam_max_cov.cpp -Wall -O2 -L./htslib/build/lib/ -I./htslib/build/include -lhts
+	g++ -std=c++11 sift_bam_max_cov.cpp -Wall -O2 -o bamsifter `pkg-config --cflags --libs htslib` -Wl,-rpath=${PREFIX}/lib
 
 
 htslib/version.h :
