--- uproc-1.2.0/common.c	2015-06-14 19:11:47.000000000 +0200
+++ uproc-1.2.0.patched/common.c	2016-07-08 15:22:45.738393909 +0200
@@ -24,9 +24,9 @@
 #include <stdlib.h>
 #include <ctype.h>
 
-#if HAVE__MKDIR
+#if HAVE_DIRECT_H
 #include <direct.h>
-#elif HAVE_MKDIR
+#else
 #include <sys/stat.h>
 #endif
 
@@ -130,9 +130,9 @@
 
 void make_dir(const char *path)
 {
-#if HAVE__MKDIR
+#if HAVE_DIRECT_H
     _mkdir(path);
-#elif HAVE_MKDIR
+#else
     mkdir(path, 0777);
 #endif
 }
