--- uproc-1.2.0/configure.ac	2015-06-23 00:03:45.774232220 +0200
+++ uproc-1.2.0.patched/configure.ac	2019-01-27 13:01:47.649944508 +0100
@@ -51,7 +51,9 @@
 AC_OPENMP
 
 # Check for the "check" unit testing library.
-PKG_CHECK_MODULES([CHECK], [check >= 0.9.4], [have_check=yes], [have_check=no])
+m4_ifdef([PKG_CHECK_MODULES], [
+    PKG_CHECK_MODULES([CHECK], [check >= 0.9.4], [have_check=yes], [have_check=no])
+])
 AM_CONDITIONAL([HAVE_CHECK], [test x$have_check = xyes])
 
 AC_ARG_ENABLE([shared],
